### ISSUE TITLE
File upload random errors

### ISSUE THREAD

**Issue by AgustinQuetto** [pipedrive/client-nodejs#142]
Hello! I'm uploading files to a deal in Pipedrive and "randomly" the api responds with error 400. Sometimes it uploads well, sometimes uploads poorly. The file is the same and is in a path. Do you know if there is any limit between file uploads?

And noteId or note_id doesn't work, apparently.

<img width="517" alt="Captura de pantalla 2020-03-20 a las 13 24 47" src="https://user-images.githubusercontent.com/15186148/77184171-31c3f100-6aae-11ea-9e2c-a20d438dfae1.png">

<img width="834" alt="Captura de pantalla 2020-03-20 a las 13 26 47" src="https://user-images.githubusercontent.com/15186148/77184769-0b528580-6aaf-11ea-8878-8f92f495a3dd.png">

**Comment by tot-ra** [comment]
Does user have a google drive integration for any chance?

**Comment by AgustinQuetto** [comment]
Google Drive is not integrated. I have the images in S3, I download them locally to upload them to Pipedrive. The file is local when it is uploaded and even if it is done with a "pre-created" one, without the previous download, the process is the same. It's usually happening with PDF files more frequently. With the images it does not usually happen.
Thank you!

**Comment by AgustinQuetto** [comment]
@tot-ra how or where can I contact support for this issue? 🤔

**Comment by AgustinQuetto** [comment]
I think I found the problem. At some point the bearer's token becomes undefined. I made a console.log of the context and the response sent appears like this. The strange thing is that it is a linear process where before a client update, treatment and some file uploads are made. It seems that the problem arises when more than one load is done at the same time.
<img width="391" alt="Captura de pantalla 2020-03-20 a las 16 26 42" src="https://user-images.githubusercontent.com/15186148/77200309-0f8b9c80-6ac9-11ea-85a9-6e991cff74a3.png">

**Comment by AgustinQuetto** [comment]
More debugs... oAuth is missing and undefined. I'm using the api token.

<img width="438" alt="Captura de pantalla 2020-03-20 a las 16 49 37" src="https://user-images.githubusercontent.com/15186148/77201253-eff57380-6aca-11ea-9aaf-31d3a84b3f3e.png">
<img width="657" alt="Captura de pantalla 2020-03-20 a las 16 49 48" src="https://user-images.githubusercontent.com/15186148/77201260-f1bf3700-6aca-11ea-9359-a4d8b0ee236e.png">

**Comment by AgustinQuetto** [comment]
It seems to be no problem. It is not listed in deals authorization. In the files I removed it and the behavior is the same.

**Comment by AgustinQuetto** [comment]
Now I have opted to use the API directly and I have no problem uploading files. I'm going to have to continue using this method without going through the library.

If I can help fix it, let me know.

<img width="774" alt="Captura de pantalla 2020-03-20 a las 17 45 21" src="https://user-images.githubusercontent.com/15186148/77205050-d0fadf80-6ad2-11ea-973d-a3c611d597fe.png">

**Comment by mykhailo-riabokon** [comment]
Hi @AgustinQuetto 

I did not succeed in reproducing this issue. This is how I've tried it

```javascript
const Pipedrive = require('pipedrive'); // latest Pipedrive Node.js 
const path = require('path');
const API_TOKEN = '<API-TOKEN>'; // replace to yours
const DEAL_ID = '<DEAL_ID>'; // replace to yours

Pipedrive.Configuration.apiToken = API_TOKEN;

function dealFileUpload(filePath) {
    return new Promise((resolve, reject) => {
        const input = {
            file: filePath,
            deal_id: DEAL_ID,
        };

        Pipedrive.FilesController.addFile(input, (err, res) => {
            if (err) {
                reject(err)
            } else {
                resolve(res);
            }
        })
    });
}


async function testIssue() {
    try {
        const imgLocalFile = path.resolve(__dirname, 'file.png'); // less than 1mb
        const pdfLocalFile = path.resolve(__dirname, 'file.pdf'); // less than 1mb
        const pdfBigLocalFile = path.resolve(__dirname, 'file_big.pdf'); // ~6mb
        const pdfRemoteFile = 'https://<domain>/file.pdf';  // link to remote file
        
        const resultImgLocal = await dealFileUpload(imgLocalFile); // - works as expected
        const resultPdfLocal = await dealFileUpload(pdfLocalFile); // - works as expected
        const resultPdBigfLocal = await dealFileUpload(pdfBigLocalFile); // - works as expected
        // const resultPdRemote = await dealFileUpload(pdfRemoteFile); // - does not work with links

    } catch (err) {
        console.log('err', err);
    }
}

testIssue()

```
I've tested it with local files (file.pdf, file.png are local files on my computer), everything works. If I try to upload a file by link, the upload will fail.

Could you provide a bit more information about:
-  the value behind `pathImage` variable you used in the last example (I mean, is it a path to a local file, Buffer, or URL)
- `.getImageS3(url)` function. What it does/returns (as I understood it's some wrapper around AWS S3)
- Node.js version `node -v`
- file size (in Mb) 

Thank you

**Comment by AgustinQuetto** [comment]
Hi! It's a local path file. getImageS3 just return the local path where the file was saved.

Node version: 12.14.1

I decided to directly use the endpoint for the upload and it works perfectly. Maybe the lib has some problem when uploading multiple files in parallel and promises are resolved with .all
It is the only factor that I found since with the direct endpoint it works wonderfully. What I don't like about the new lib is that it is not instantiable and is somewhat annoying when handling multiple clients and different concurrent tokens.
If you like, I could abstract the code that was having problems and upload it to a gist.
